tests/rust_usage_test/tests/integration_test.rs

```diff
@@ -1747,7 +1776,14 @@ mod follow_impls {
 
         let buf: Vec<u8> = vec![1, 0, 0, 0, /* struct data */ 1, 2, 3, 4];
         let fs: flatbuffers::FollowStart<flatbuffers::Vector<FooStruct>> = flatbuffers::FollowStart::new();
-        assert_eq!(fs.self_follow(&buf[..], 0).safe_slice(), &vec![FooStruct{a: 1, b: 2, c: 1027}][..]);
+        #[cfg(target_endian = "little")]
+        {
+            assert_eq!(fs.self_follow(&buf[..], 0).safe_slice(), &vec![FooStruct{a: 1, b: 2, c: 1027}][..]);
```

Comment:
I think the real solution here is to use the struct constructor function, not to create it with the default initializer, like I seem to have done here.

Reply:
So, are you thinking that we should write a `Push` implementation for `FooStruct` in these 3 tests (`follow_impls::to_struct`, `follow_impls::to_slice_of_struct_elements`, and `follow_impls::to_vector_of_struct_elements`), and then construct this as a FlatBuffers struct?

Reply:
@tymcauley Nothing as complicated as that -- just to use a `new` function, like we do with `Vec3` here: https://github.com/google/flatbuffers/blob/master/tests/monster_test_generated.rs#L556

Reply:
Sounds great, see if you think 6c1d51f resolves this.